src/trainers/mitie_sklearn_trainer.py

```diff
@@ -1,24 +1,37 @@
 from mitie import *
-import os
+import cloudpickle
 import datetime
 import json
+import os
+
+from rasa_nlu.classifiers.sklearn_intent_classifier import SklearnIntentClassifier
+from rasa_nlu.featurizers.mitie_featurizer import MITIEFeaturizer
+from rasa_nlu.tokenizers.mitie_tokenizer import MITIETokenizer
+from rasa_nlu.trainers.trainer import Trainer
 from training_utils import write_training_metadata
 
 
-class MITIESklearnTrainer(object):
-    def __init__(self, config):
-        self.name = "mitie"
+class MITIESklearnTrainer(Trainer):
+    SUPPORTED_LANGUAGES = {"en"}
+
+    def __init__(self, fe_file, language_name, max_num_threads=1):
+        self.name = "mitie_sklearn"
         self.training_data = None
         self.intent_classifier = None
         self.entity_extractor = None
         self.training_data = None
-        self.fe_file = config.mitie_file
-        self.feature_extractor = total_word_feature_extractor(self.fe_file)
+        self.fe_file = fe_file
+        self.featurizer = MITIEFeaturizer(self.fe_file)
+        self.max_num_threads = max_num_threads
+        self.ensure_language_support(language_name)
 
-    def train(self, data):
+    def train(self, data, test_split_size=0.1):
```

Comment:
yep that's cleaner. We should add abstract methods for the `train_intent_classifier` & `train_entity_extractor`  to the base class as well.